maven-core/src/main/java/org/apache/maven/graph/ProjectCollectionStrategy.java

```diff
@@ -0,0 +1,41 @@
+package org.apache.maven.graph;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.execution.MavenExecutionRequest;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.project.ProjectBuildingException;
+
+import java.util.List;
+
+/**
+ * Describes strategies for finding projects that Maven could build.
+ */
+public interface ProjectCollectionStrategy
+{
+    /**
+     *
+     * @param request
+     * @return
+     * @throws ProjectBuildingException
+     */
+    List<MavenProject> collectProjects( final MavenExecutionRequest request )
+            throws ProjectBuildingException;
```

Comment:
Shouldn't this be an exception in the same package?

Reply:
I've moved the Collector related classes to a new package within the `project` package. Is that better?

Reply:
I think that was indeed the best option (new exception versus move to project package)